color/src/main/java/com/afollestad/materialdialogs/color/DialogColorChooserExt.kt

```diff
@@ -72,3 +93,135 @@ fun MaterialDialog.colorChooser(
+    }
+    val gridRecyclerView : RecyclerView = dialog.getCustomView()!!.findViewById(R.id.rvGrid)
+    val gridColumnCount = dialog.windowContext.resources
+            .getInteger(R.integer.color_grid_column_count)
+    gridRecyclerView.layoutManager = GridLayoutManager(
+            dialog.windowContext, gridColumnCount
+    )
+
+    val adapter = ColorGridAdapter(
+            dialog = dialog,
+            colors = colors,
+            subColors = subColors,
+            initialSelection = initialSelection,
+            waitForPositiveButton = waitForPositiveButton,
+            callback = selection
+    )
+    gridRecyclerView.adapter = adapter
+}
+
+private fun updateCustomPage(dialog: MaterialDialog, supportCustomAlpha: Boolean, @ColorInt initialSelection: Int?, waitForPositiveButton: Boolean, selection: ColorCallback) {
+  val customPage: View = dialog.getPageCustomView()
+  val vColor: View = customPage.findViewById(R.id.v_color)
+  val llAlpha: LinearLayout = customPage.findViewById(R.id.llAlpha)
+  val sbAlpha: SeekBar = customPage.findViewById(R.id.sb_alpha)
+  val sbRed: SeekBar = customPage.findViewById(R.id.sb_red)
+  val sbGreen: SeekBar = customPage.findViewById(R.id.sb_green)
+  val sbBlue: SeekBar = customPage.findViewById(R.id.sb_blue)
+  val tvAlphaValue: TextView = customPage.findViewById(R.id.tv_alpha_value)
+  val tvRedValue: TextView = customPage.findViewById(R.id.tv_red_value)
+  val tvGreenValue: TextView = customPage.findViewById(R.id.tv_green_value)
+  val tvBlueValue: TextView = customPage.findViewById(R.id.tv_blue_value)
+
+  sbAlpha.tint(Color.BLACK)
+  sbRed.tint(Color.RED)
+  sbGreen.tint(Color.GREEN)
+  sbBlue.tint(Color.BLUE)
+
+  initialSelection?.let {
+    if (supportCustomAlpha) {
+      sbAlpha.progress = Color.alpha(it)
+    }
+    sbRed.progress = Color.red(it)
+    sbGreen.progress = Color.green(it)
+    sbBlue.progress = Color.blue(it)
+  } ?: run {
+    sbAlpha.progress = 255
+  }
+
+  llAlpha.visibility = if (supportCustomAlpha) View.VISIBLE else View.GONE
+
+  val listener = object: SeekBar.OnSeekBarChangeListener {
+    override fun onProgressChanged(p0: SeekBar?, p1: Int, p2: Boolean) {
+      onCustomValueChanged(dialog, supportCustomAlpha, waitForPositiveButton, true, customPage, vColor, sbAlpha, sbRed, sbGreen, sbBlue, tvAlphaValue, tvRedValue, tvGreenValue, tvBlueValue, selection)
+    }
+
+    override fun onStartTrackingTouch(p0: SeekBar?) {}
+    override fun onStopTrackingTouch(p0: SeekBar?) { }
+  }
+
+  sbAlpha.setOnSeekBarChangeListener(listener)
+  sbRed.setOnSeekBarChangeListener(listener)
+  sbGreen.setOnSeekBarChangeListener(listener)
+  sbBlue.setOnSeekBarChangeListener(listener)
+
+  onCustomValueChanged(dialog, supportCustomAlpha, waitForPositiveButton, initialSelection != null, customPage, vColor, sbAlpha, sbRed, sbGreen, sbBlue, tvAlphaValue, tvRedValue, tvGreenValue, tvBlueValue, selection)
```

Comment:
Kotlin style guides will want each of these arguments on separate lines - better yet, named. Unless the method signature can be reduced.